From df55b1f16ea2fb5e56f0fe69419904d50b7e2476 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Pavel=20B=C5=99ezina?= <pbrezina@redhat.com>
Date: Wed, 12 Oct 2022 14:32:31 +0200
Subject: [PATCH 4/6] confdb: chande debug level when no domain are found in
 confdb_get_domains
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

We print the error as fatal error in the monitor to make sure the
message is correctly visible. However, the error is not fatal for tools
like sss_cache and it should not be printed there by default.

Since the tools have default debug level set to SSSDBG_FATAL_FAILURE, it
is sufficient to just drop the level to critical.

Resolves: https://github.com/SSSD/sssd/issues/6387

Reviewed-by: Alejandro López <allopez@redhat.com>
Reviewed-by: Sumit Bose <sbose@redhat.com>
---
 src/confdb/confdb.c   | 5 ++---
 src/monitor/monitor.c | 5 +++--
 2 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/src/confdb/confdb.c b/src/confdb/confdb.c
index 3ecdaa3b91e3b550f670768d336e9d7d85bb66e6..cf0476f759827f3e1aa1f445d4e71c09b3ec3239 100644
--- a/src/confdb/confdb.c
+++ b/src/confdb/confdb.c
@@ -1796,11 +1796,10 @@ int confdb_get_domains(struct confdb_ctx *cdb,
 
     ret = confdb_get_enabled_domain_list(cdb, tmp_ctx, &domlist);
     if (ret == ENOENT) {
-        DEBUG(SSSDBG_FATAL_FAILURE, "No domains configured, fatal error!\n");
+        DEBUG(SSSDBG_CRIT_FAILURE, "No domains configured, fatal error!\n");
         ret = ERR_NO_DOMAIN_ENABLED;
         goto done;
-    }
-    if (ret != EOK) {
+    } else if (ret != EOK) {
         DEBUG(SSSDBG_FATAL_FAILURE, "Fatal error retrieving domains list!\n");
         goto done;
     }
diff --git a/src/monitor/monitor.c b/src/monitor/monitor.c
index 511e13971d253bc25cb9a04008c1a363f2182748..2196c2e0b0fdd630335333e7d53026e34302e512 100644
--- a/src/monitor/monitor.c
+++ b/src/monitor/monitor.c
@@ -2564,9 +2564,10 @@ int main(int argc, const char *argv[])
             break;
         default:
             DEBUG(SSSDBG_FATAL_FAILURE,
-                 "SSSD couldn't load the configuration database.\n");
+                 "SSSD couldn't load the configuration database [%d]: %s\n",
+                 ret, sss_strerror(ret));
             sss_log(SSS_LOG_CRIT,
-                   "SSSD couldn't load the configuration database [%d]: %s.\n",
+                   "SSSD couldn't load the configuration database [%d]: %s\n",
                     ret, sss_strerror(ret));
             break;
         }
-- 
2.37.3

